pkg/istio-agent/sds-agent.go

```diff
@@ -207,11 +206,7 @@ func NewAgent(proxyConfig *mesh.ProxyConfig, cfg *AgentConfig, sopts *security.O
 		sa.secOpts.TLSEnabled = false
 	}
 	// If proxy is using file mounted certs, JWT token is not needed.
-	if sa.secOpts.FileMountedCerts {
-		sa.secOpts.UseLocalJWT = false
-	} else {
-		sa.secOpts.UseLocalJWT = sa.CertsPath == "" // true if we don't have a key.pem
-	}
+	sa.secOpts.UseLocalJWT = !sa.FileMountedCerts
```

Comment:
Could you add https://github.com/istio/istio/blob/master/pkg/istio-agent/sds-agent_test.go and add a test that verifies the fields in sa returned from NewAgent() are correct? It would be nice to have a test covering these fields so this fix would not easily break by other PRs.

Reply:
Added a test.